src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Calendar/Calendar.shared.cs

```diff
@@ -0,0 +1,533 @@
+		}
+
+		static void OnFirstDayOfWeekChanged(BindableObject bindable, object oldValue, object newValue)
+		{
+			var calendar = (Calendar)bindable;
+			calendar.UpdateWeekDayHeaders();
+			calendar.UpdateCalendarDays();
+		}
+
+		/// <summary>
+		/// Backing BindableProperty for the <see cref="WeekDayHeaderControlTemplate"/> property.
+		/// </summary>
+		public static readonly BindableProperty WeekDayHeaderControlTemplateProperty =
+			BindableProperty.Create(nameof(WeekDayHeaderControlTemplateProperty), typeof(ControlTemplate),
+				typeof(Calendar), propertyChanged: OnWeekDayHeaderControlTemplateChanged);
+
+		/// <summary>
+		/// Gets or sets <see cref="ControlTemplate"/> for week day header at the top of <see cref="Calendar"/>.
+		/// </summary>
+		public ControlTemplate? WeekDayHeaderControlTemplate
+		{
+			get => (ControlTemplate?)GetValue(WeekDayHeaderControlTemplateProperty);
+			set => SetValue(WeekDayHeaderControlTemplateProperty, value);
+		}
+
+		static void OnWeekDayHeaderControlTemplateChanged(BindableObject bindable, object oldValue, object newValue)
+		{
+			var calendar = (Calendar)bindable;
+			calendar.UpdateWeekDayHeaders();
+		}
+
+		/// <summary>
+		/// Backing BindableProperty for the <see cref="DayControlTemplate"/> property.
+		/// </summary>
+		public static readonly BindableProperty DayControlTemplateProperty =
+			BindableProperty.Create(nameof(DayControlTemplate), typeof(ControlTemplate), typeof(Calendar),
+				propertyChanged: OnDayControlTemplateChanged);
+
+		/// <summary>
+		/// Gets or sets <see cref="ControlTemplate"/> for <see cref="CalendarDay"/>.
+		/// </summary>
+		public ControlTemplate? DayControlTemplate
+		{
+			get => (ControlTemplate?)GetValue(DayControlTemplateProperty);
+			set => SetValue(DayControlTemplateProperty, value);
+		}
+
+		static void OnDayControlTemplateChanged(BindableObject bindable, object oldValue, object newValue)
+		{
+			var calendar = (Calendar)bindable;
+			calendar.UpdateCalendarDays();
+		}
+
+		/// <summary>
+		/// Backing BindableProperty for the <see cref="SelectedDays"/> property.
+		/// </summary>
+		public static readonly BindableProperty SelectedDaysProperty =
+			BindableProperty.Create(nameof(SelectedDays), typeof(IReadOnlyCollection<DateTime>), typeof(Calendar), Enumerable.Empty<DateTime>().ToList(), propertyChanged: OnSelectedDaysChanged);
+
+		/// <summary>
+		/// Gets or sets selected days.
+		/// </summary>
+		public IReadOnlyCollection<DateTime> SelectedDays
+		{
+			get => (IReadOnlyCollection<DateTime>)GetValue(SelectedDaysProperty);
+			set => SetValue(SelectedDaysProperty, value);
+		}
+
+		static void OnSelectedDaysChanged(BindableObject bindable, object oldValue, object newValue)
+		{
+			var daysToSelect = (IList<DateTime>)newValue;
+			var calendar = (Calendar)bindable;
+
+			if (calendar.SelectionMode is CalendarSelectionMode.SingleSelect
+				&& daysToSelect.Count > 1)
+			{
+				throw new InvalidOperationException(
```

Comment:
Do we want to throw an exception? If we throw an exception here, I believe that dev can't catch, mainly if it's using XAML, in other words the app will crash, isn't better to log this message and just return?

Reply:
I support throwing the exception because it means that the code has gotten to an invalid state; a developer should never be in a scenario where `calendar.SelectionMode is CalendarSelectionMode.SingleSelect &&  daysToSelect.Count > 1`